### PR TITLE
feat: Separate mass and volume residuals for output in compositional flow solver

### PR DESCRIPTION
**Pull request by paveltomin**

### CODE REVIEW

File: src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp [hunk 1]
```diff
@@ -433,31 +436,37 @@ real64 CompositionalMultiphaseFVM::calculateResidualNorm( real64 const & GEOS_UN
...
-      solverBaseKernels::L2ResidualNormHelper::computeGlobalNorm( localResidualNorm[0], localResidualNormalizer[0], residualNorm );
+      solverBaseKernels::L2ResidualNormHelper::
+        computeGlobalNorm( localResidualNorm, localResidualNormalizer, globalResidualNorm );
     }
+    residualNorm = sqrt( globalResidualNorm[0] * globalResidualNorm[0] + globalResidualNorm[1] * globalResidualNorm[1] );
 
     if( getLogLevel() >= 1 && logger::internal::rank == 0 )
     {
-      std::cout << GEOS_FMT( "        ( R{} ) = ( {:4.2e} )", coupledSolverAttributePrefix(), residualNorm );
+      std::cout << GEOS_FMT( "        ( Rmass Rvol ) = ( {:4.2e} {:4.2e} )",
```
Comment (CusiniM) [inline src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp:465]: could use the `GEOS_LOG_RANK_0` macro for this.

**Review by CusiniM — APPROVED** [review]

File: src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp (hunk 1 above)
Comment (paveltomin) [reply src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp:465]: It adds end of line and those residuals are collected from all solvers and printed in one line

File: src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp [hunk 2]
```diff
@@ -433,31 +436,37 @@ real64 CompositionalMultiphaseFVM::calculateResidualNorm( real64 const & GEOS_UN
       solverBaseKernels::L2ResidualNormHelper::
         computeGlobalNorm( localResidualNorm, localResidualNormalizer, globalResidualNorm );
     }
-    residualNorm = sqrt( globalResidualNorm[0] * globalResidualNorm[0] + globalResidualNorm[1] * globalResidualNorm[1] );
+    residualNorm = sqrt( globalResidualNorm[0] * globalResidualNorm[0] + globalResidualNorm[1] * globalResidualNorm[1]  + globalResidualNorm[1] * globalResidualNorm[1] );
```
Comment (rasimHZ) [inline src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp]: Is this a typo? The third term should be  globalResidualNorm[2] * globalResidualNorm[2]

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp]: yes thanks

**Comment by paveltomin** [comment]
rebaseline needed because of some numerical diff causing one iteration more for some cases, e.g.:
```
Error: /Problem/Solvers/compflow/SolverStatistics/numSuccessfulLinearIterations/__values__
	Scalar values of types int32 and int32 differ: 548, 547.
```

File: src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp [hunk 3]
```diff
@@ -328,7 +328,7 @@ real64 CompositionalMultiphaseFVM::calculateResidualNorm( real64 const & GEOS_UN
 {
   GEOS_MARK_FUNCTION;
 
-  integer constexpr numNorm = 2; // mass balance and energy balance
+  integer constexpr numNorm = 3; // mass/volume balance and energy balance
```
Comment (rrsettgast) [inline src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp:331]: @paveltomin Is the third residual slot for thermal?

**Review by rrsettgast — APPROVED** [review]

**Comment by rrsettgast** [comment]
@paveltomin Have you checked the baselines to ensure there are no substantiative changes?

File: src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp (hunk 3 above)
Comment (paveltomin) [reply src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp:331]: yes

**Comment by paveltomin** [comment]
> @paveltomin Have you checked the baselines to ensure there are no substantiative changes?

yes, i did check and didn't see any solution changes
few cases with very tight convergence settings got +-1 newton iterations, i believe just because of numerical diffs when residual value is around convergence tolerance

**Pull request merged**